src/dotnet/Program.cs

```diff
@@ -138,6 +141,16 @@ internal static int ProcessArgs(string[] args, ITelemetry telemetryClient)
 
         }
 
+        /// <summary>
+        /// Performs initialization that must be done before anything else in the process.
+        /// </summary>
+        private static void InitializeProcess()
+        {
+            // by default, .NET Core doesn't have all code pages needed for Console apps.
+            // see the .NET Core Notes in https://msdn.microsoft.com/en-us/library/system.diagnostics.process(v=vs.110).aspx
+            Encoding.RegisterProvider(CodePagesEncodingProvider.Instance);
```

Comment:
>  I'm not entirely sure why it is actually needed here as it should be needed to support Unicode encodings.

On my Windows machine, the default code page is:

```
C:\cli>chcp
Active code page: 437
```

Which is not in the default .NET Core code page list.  Similarly, users can set their console code page to whatever they want, and we should be able to work with it, if possible.

Also, another thing we do in the CLI is launch separate processes, and then forward the process's standard error and out to our process, and interpret it. One example of this is when we run `csc.exe`, we Regex the output looking for compiler warnings and errors so we can count them up at the end. Another is when we launch the user's app, we forward the output to our output during `dotnet run`. The separate process can use whatever code page it wants to write to its standard out. If we don't have the correct code page, we won't be reading the results correctly.

Looking at my machine, the System.Text.Encoding.CodePages.dll is 852 **K**B.

/cc @tarekgh